src/Microsoft.ML.StandardLearners/Standard/SdcaMultiClass.cs

```diff
@@ -47,14 +46,55 @@ public sealed class Arguments : ArgumentsBase
         private readonly Arguments _args;
 
         public override PredictionKind PredictionKind => PredictionKind.MultiClassClassification;
+        protected override SchemaShape.Column[] OutputColumns { get; }
 
-        public SdcaMultiClassTrainer(IHostEnvironment env, Arguments args)
-            : base(args, env, LoadNameValue)
+        public SdcaMultiClassTrainer(IHostEnvironment env, Arguments args,
+            string featureColumn, string labelColumn, string weightColumn = null)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(LoadNameValue), args, MakeFeatureColumn(featureColumn), MakeLabelColumn(labelColumn), MakeWeightColumn(weightColumn))
         {
             _loss = args.LossFunction.CreateComponent(env);
-            base.Loss = _loss;
-            NeedShuffle = args.Shuffle;
+            Loss = _loss;
             _args = args;
+            OutputColumns = new[]
+            {
+                new SchemaShape.Column(DefaultColumnNames.Score, SchemaShape.Column.VectorKind.Vector, DataKind.R4, false),
+                new SchemaShape.Column(DefaultColumnNames.PredictedLabel, SchemaShape.Column.VectorKind.Scalar, DataKind.U4, true)
+            };
+        }
```

Comment:
Hmmm. We'll want to factor that out as we acquire more of these I suppose... plus of course we'll need to push the scorer somewhere else.

Reply:
Indeed

---
In reply to: [212448831](https://github.com/dotnet/machinelearning/pull/716#discussion_r212448831) [](ancestors = 212448831)